packages/router/utils.ts

```diff
@@ -1351,12 +1351,26 @@ export class DeferredData {
       this.unlistenAbortSignal();
     }
 
-    if (error) {
+    // use argument length to detect error or value resolution
+    if (arguments.length === 3) {
       Object.defineProperty(promise, "_error", { get: () => error });
       this.emit(false, key);
       return Promise.reject(error);
```

Comment:
What if we just handled `resolve(undefined)` and `reject(undefined)` together ahead of both of these and skipped the `arguments.length` check?

```js
if (error === undefined && data == undefined) {
  let undefinedError = new Error(
    `Deferred data for key "${key}" resolved/rejected with \`undefined\`, ` +
    `you must resolve/reject with a value or \`null\`.`
  );
  Object.defineProperty(promise, "_error", { get: () => undefinedError });
  this.emit(false, key);
  return Promise.reject(undefinedError);
}
```

Reply:
I like that.